File: FlashVoteBackend/Models/Poll.cs
namespace FlashVoteBackend.Models
{
    public record Poll
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime? ExpiresAt { get; set; }
        public bool IsClosed { get; set; }
        public List<Option> Options { get; set; } = new();
    }
}

File: FlashVoteFrontend/src/types/poll.ts
export interface PollOption {
  id: string
  text: string
  voteCount: number
  pollId: string
}

export interface PollListItem {
  id: string
  title: string
  description: string
  createdAt: string
  expiresAt: string | null
  isClosed: boolean
}

export interface PollDetail extends PollListItem {
  options: PollOption[]
}

export interface VoteParams {
  pollId: string
  optionId: string
}